modules/400-descheduler/templates/configmap.yaml

```
@@ -1,111 +1,128 @@
-              "memory": 50
-              "pods": 40
-            targetThresholds:
-              "cpu" : 80
-              "memory": 90
-              "pods": 80
-{{- else }}
+          {{- $strategies.lowNodeUtilization | toYaml | nindent 10 }}
+    {{- end }}
+  {{- else }}
       "LowNodeUtilization":
         enabled: false
-{{- end }}
+  {{- end }}
 
-{{- if .Values.descheduler.highNodeUtilization }}
+  {{- if hasKey $strategies "highNodeUtilization" }}
       "HighNodeUtilization":
         enabled: true
+    {{- if $strategies.highNodeUtilization  }}
         params:
-          nodeFit: true
-          nodeResourceUtilizationThresholds:
-            thresholds:
-              "cpu" : 50
-              "memory": 50
-{{- else }}
+          {{- $strategies.highNodeUtilization | toYaml | nindent 10 }}
+    {{- end }}
+  {{- else }}
       "HighNodeUtilization":
         enabled: false
-{{- end }}
+  {{- end }}
 
-{{- if .Values.descheduler.removePodsViolatingNodeTaints }}
+  {{- if hasKey $strategies "removePodsViolatingNodeTaints" }}
       "RemovePodsViolatingNodeTaints":
         enabled: true
-{{- else }}
+    {{- if $strategies.removePodsViolatingNodeTaints  }}
+        params:
+          {{- $strategies.removePodsViolatingNodeTaints | toYaml | nindent 10 }}
+    {{- end }}
+  {{- else }}
       "RemovePodsViolatingNodeTaints":
         enabled: false
-{{- end }}
+  {{- end }}
 
-{{- if .Values.descheduler.removePodsViolatingTopologySpreadConstraint }}
+  {{- if hasKey $strategies "removePodsViolatingTopologySpreadConstraint" }}
       "RemovePodsViolatingTopologySpreadConstraint":
         enabled: true
-{{- else }}
+    {{- if $strategies.removePodsViolatingTopologySpreadConstraint  }}
+        params:
+          {{- $strategies.removePodsViolatingTopologySpreadConstraint | toYaml | nindent 10 }}
+    {{- end }}
+  {{- else }}
       "RemovePodsViolatingTopologySpreadConstraint":
         enabled: false
-{{- end }}
+  {{- end }}
 
-{{- if .Values.descheduler.removePodsHavingTooManyRestarts }}
+  {{- if hasKey $strategies "removePodsHavingTooManyRestarts" }}
       "RemovePodsHavingTooManyRestarts":
         enabled: true
+    {{- if $strategies.removePodsHavingTooManyRestarts  }}
         params:
-          podsHavingTooManyRestarts:
-            podRestartThreshold: 100
-            includingInitContainers: true
-{{- else }}
+          {{- $strategies.removePodsHavingTooManyRestarts | toYaml | nindent 10 }}
+    {{- end }}
+  {{- else }}
       "RemovePodsHavingTooManyRestarts":
         enabled: false
-{{- end }}
+  {{- end }}
 
-{{- if .Values.descheduler.podLifeTime }}
+  {{- if hasKey $strategies "podLifeTime" }}
       "PodLifeTime":
         enabled: true
+    {{- if $strategies.podLifeTime  }}
         params:
-          podLifeTime:
-            maxPodLifeTimeSeconds: 86400
-            podStatusPhases:
-            - "Pending"
-{{- else }}
+          {{- $strategies.podLifeTime | toYaml | nindent 10 }}
```

```suggestion
          params: {{- $strategies.podLifeTime | toYaml | nindent 10 }}
```
NIT: `nindent` adds new line, so you do not need to do it manually.

---

Done.